### DISCUSSION TITLE
AX.25/APRS Help

### DISCUSSION THREAD

**Discussion by n9wxu** [jgromes/RadioLib#560] [General]
I am trying to enable AFSK modulation so I can send APRS data.  I am using an SX1278 so it is a supported transceiver.
When I use a Heltec ESP32 LoRa V2 and use pin 12 for DIO2, it works fine.
When I use an Adafruit Feather LoRa M0 (SAMD21) I get noise.
I have tried D5 and D9 as the DIO2 pin.  I can get a tone (so morse works) but when I put a scope on the pin, the frequency is not consistent even for the morse tone.  I have attached a sample of the DIO2 signal from the Feather.
<img width="1002" alt="image" src="https://user-images.githubusercontent.com/750641/186327842-18c562c1-9b6e-4ea7-8cf4-98bdc908dc74.png">

**Comment by jgromes** [comment]
As far as I can tell RadioLib is building correctly for Feather M0 (i.e. the platform is correctly detected as Adafruit SAMD). Internally, to generate tones for AFSK, it only calls the tone() function provided by the platform, so I would suggest just calling tone() from the Arduino sketch to check if the frequency is still inconsistent. If so, then it's an issue with the implementation of tone().
